lib/util/Components.js

```diff
@@ -258,6 +258,8 @@ function componentRule(rule, context) {
       var returnsReactCreateElement =
         node[property] &&
         node[property].callee &&
+        node[property].callee.object &&
+        node[property].callee.object.name === 'React' &&
```

Comment:
I'll just temporarily add in a disable comment so I can feel good about pushing it up. I'll remove it after receiving feedback.